### PR TITLE
Spring Security upgrade

### PR DESCRIPTION
**Pull request by jglick**
Description of a major migration centered around https://github.com/jenkinsci/jenkins/pull/4848.

"Submitted for approval as draft JEP by @jenkinsci/jep-editors.

### CODE REVIEW

**Comment by jglick** [comment]
@rwinch FYI

**Comment by jglick** [comment]
@jenkinsci/jep-editors?

**Review by rtyler — APPROVED** [review]
From a document standpoint, this looks great @jglick 

Let me slap a number on this and i'll merge it as a Draft

File: jep/227/compatibility.adoc [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+|link:https://plugins.jenkins.io/google-cloud-health-check/[google-cloud-health-check]
+|Probably compatible
+|link:https://github.com/jenkinsci/google-cloud-health-check-plugin/blob/a6b68119ebd64c1bad026ad5ed7a04c964bcb382/src/test/java/com/google/jenkins/plugins/health/HealthCheckActionTest.java#L63[This code]
+would need to be adjusted to make tests pass.
+
+|link:https://plugins.jenkins.io/google-login/[google-login]
+|Probably compatible
+|Untested.
+
+|link:https://plugins.jenkins.io/jenkins-cloudfoundry-uaa/[jenkins-cloudfoundry-uaa]
```
Comment (daniel-beck) [inline jep/227/compatibility.adoc:147]: This link doesn't work, the plugin doesn't exist. Is this https://plugins.jenkins.io/cloudfoundry/ ? Superficially perhaps not, couldn't find a `BeanBuilder` there at a glance.

Comment (jglick) [reply jep/227/compatibility.adoc:147]: #308

**Pull request merged**